pkg/controller/kieapp/defaults/defaults_test.go

```diff
@@ -6805,3 +6805,232 @@ func TestDashbuilderWithPartialCORS(t *testing.T) {
+	envs := env.Servers[0].DeploymentConfigs[0].Spec.Template.Spec.Containers[0].Env
+	strategyEnabled, strategyEnabledTwo, strategyEnabledThree := checkEnvsKieserverStartupStrategy(t, envs, api.OpenshiftStartupStrategy, ttl)
+	assert.True(t, strategyEnabled)
+	assert.True(t, strategyEnabledTwo)
+	assert.True(t, strategyEnabledThree)
+}
+
+func checkEnvsKieserverStartupStrategy(t *testing.T, envs []corev1.EnvVar, strategy string, ttl string) (bool, bool, bool) {
+	strategyEnabled := false
+	strategyEnabledTwo := false
+	strategyEnabledThree := false
+	if ttl == "0" {
+		strategyEnabledThree = true
+	}
+	if strategy == api.OpenshiftStartupStrategy {
+		strategyEnabledTwo = true
+	}
+	for _, env := range envs {
+		if strings.HasPrefix(env.Name, "KIE_SERVER_STARTUP_STRATEGY") {
+			assert.Equal(t, env.Value, strategy)
+			strategyEnabled = true
+		}
+
+		if strings.HasPrefix(env.Name, "KIE_SERVER_CONTROLLER_OPENSHIFT_ENABLED") {
+			assert.Equal(t, env.Value, "false")
+			strategyEnabledTwo = true
+		}
+
+		if strings.HasPrefix(env.Name, "KIE_SERVER_CONTROLLER_TEMPLATE_CACHE_TTL") {
+			assert.Equal(t, env.Value, ttl)
+			strategyEnabledThree = true
+		}
+	}
+	return strategyEnabled, strategyEnabledTwo, strategyEnabledThree
+}
+
+func TestKieServerControllerStartupStrategyConfiguration(t *testing.T) {
+	cr := &api.KieApp{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "test",
+		},
+		Spec: api.KieAppSpec{
+			Environment: api.RhpamProductionImmutable,
+			CommonConfig: api.CommonConfig{
+				StartupStrategy: createStartupStrategy(api.ControllerStartupStrategy, 0),
+			},
+			Objects: api.KieAppObjects{
+				Servers: []api.KieServerSet{},
+			},
+		},
+	}
+
+	env, err := GetEnvironment(cr, test.MockService())
+	assert.Nil(t, err, "Error getting prod environment")
+	assert.NotNil(t, env)
+	envs := env.Servers[0].DeploymentConfigs[0].Spec.Template.Spec.Containers[0].Env
+
+	strategyEnabled, strategyEnabledTwo, strategyEnabledThree := checkEnvsKieserverStartupStrategy(t, envs, api.ControllerStartupStrategy, "0")
+	assert.True(t, strategyEnabled)
+	assert.True(t, strategyEnabledTwo)
+	assert.True(t, strategyEnabledThree)
+}
+
+func TestConsoleDefaultStartupStrategyConfiguration(t *testing.T) {
+	cr := &api.KieApp{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "test",
+		},
+		Spec: api.KieAppSpec{
+			Environment:  api.RhpamAuthoring,
+			CommonConfig: api.CommonConfig{},
+			Objects: api.KieAppObjects{
+				Console: &api.ConsoleObject{},
+			},
+		},
+	}
+	checkConsoleOpenshiftStrategyAssertions(t, cr, true)
+}
+
+func checkConsoleControllerStrategyAssertions(t *testing.T, cr *api.KieApp) {
+	env, err := GetEnvironment(cr, test.MockService())
+	assert.Nil(t, err, "Error getting prod environment")
+	assert.NotNil(t, env)
+	envs := env.Console.DeploymentConfigs[0].Spec.Template.Spec.Containers[0].Env
+	controllerStrategyEnabled := false
+	for _, env := range envs {
+
+		if strings.HasPrefix(env.Name, "KIE_SERVER_CONTROLLER_OPENSHIFT_ENABLED") {
```

Comment:
why this for loop here?
you can just check the env using the getVariable fund and assert the result to false.

Reply:
ok